### PR TITLE
linux_xanmod-rt: init at 5.13.1

### PR DESCRIPTION
**Pull request by moni-dz**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by fufexan — APPROVED** [review]

**Comment by Kranzes** [comment]
Result of `nixpkgs-review pr 133511` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages_xanmod_rt.anbox</li>
    <li>linuxPackages_xanmod_rt.ixgbevf</li>
    <li>linuxPackages_xanmod_rt.mxu11x0</li>
    <li>linuxPackages_xanmod_rt.ndiswrapper</li>
    <li>linuxPackages_xanmod_rt.phc-intel</li>
    <li>linuxPackages_xanmod_rt.r8125</li>
    <li>linuxPackages_xanmod_rt.rtl8723bs</li>
    <li>linuxPackages_xanmod_rt.sch_cake</li>
    <li>linuxPackages_xanmod_rt.tbs</li>
  </ul>
</details>
<details>
  <summary>12 packages failed to build:</summary>
  <ul>
    <li>linuxPackages_xanmod_rt.amdgpu-pro</li>
    <li>linuxPackages_xanmod_rt.broadcom_sta</li>
    <li>linuxPackages_xanmod_rt.ena</li>
    <li>linuxPackages_xanmod_rt.nvidia_x11</li>
    <li>linuxPackages_xanmod_rt.nvidia_x11_beta</li>
    <li>linuxPackages_xanmod_rt.nvidia_x11_legacy340</li>
    <li>linuxPackages_xanmod_rt.nvidia_x11_legacy390</li>
    <li>linuxPackages_xanmod_rt.nvidia_x11_vulkan_beta</li>
    <li>linuxPackages_xanmod_rt.openafs (linuxPackages_xanmod_rt.openafs_1_8)</li>
    <li>linuxPackages_xanmod_rt.openafs_1_9</li>
    <li>linuxPackages_xanmod_rt.perf</li>
    <li>linuxPackages_xanmod_rt.zfs (linuxPackages_xanmod_rt.zfsStable ,linuxPackages_xanmod_rt.zfsUnstable)</li>
  </ul>
</details>
<details>
  <summary>73 packages built:</summary>
  <ul>
    <li>linuxPackages_xanmod_rt.acpi_call</li>
    <li>linuxPackages_xanmod_rt.akvcam</li>
    <li>linuxPackages_xanmod_rt.apfs</li>
    <li>linuxPackages_xanmod_rt.asus-wmi-sensors</li>
    <li>linuxPackages_xanmod_rt.batman_adv</li>
    <li>linuxPackages_xanmod_rt.bbswitch</li>
    <li>linuxPackages_xanmod_rt.bcc</li>
    <li>linuxPackages_xanmod_rt.bpftrace</li>
    <li>linuxPackages_xanmod_rt.can-isotp</li>
    <li>linuxPackages_xanmod_rt.chipsec</li>
    <li>linuxPackages_xanmod_rt.cpupower</li>
    <li>linuxPackages_xanmod_rt.cryptodev</li>
    <li>linuxPackages_xanmod_rt.ddcci-driver</li>
    <li>linuxPackages_xanmod_rt.digimend</li>
    <li>linuxPackages_xanmod_rt.dpdk-kmods</li>
    <li>linuxPackages_xanmod_rt.evdi</li>
    <li>linuxPackages_xanmod_rt.facetimehd</li>
    <li>linuxPackages_xanmod_rt.fwts-efi-runtime</li>
    <li>linuxPackages_xanmod_rt.gcadapter-oc-kmod</li>
    <li>linuxPackages_xanmod_rt.hid-nintendo</li>
    <li>linuxPackages_xanmod_rt.hyperv-daemons</li>
    <li>linuxPackages_xanmod_rt.intel-speed-select</li>
    <li>linuxPackages_xanmod_rt.isgx</li>
    <li>linuxPackages_xanmod_rt.it87</li>
    <li>linuxPackages_xanmod_rt.jool</li>
    <li>linux_xanmod-rt (linuxPackages_xanmod_rt.kernel)</li>
    <li>linuxPackages_xanmod_rt.kvmfr</li>
    <li>linuxPackages_xanmod_rt.lttng-modules</li>
    <li>linuxPackages_xanmod_rt.mba6x_bl</li>
    <li>linuxPackages_xanmod_rt.mbp2018-bridge-drv</li>
    <li>linuxPackages_xanmod_rt.mwprocapture</li>
    <li>linuxPackages_xanmod_rt.netatop</li>
    <li>linuxPackages_xanmod_rt.nvidiabl</li>
    <li>linuxPackages_xanmod_rt.oci-seccomp-bpf-hook</li>
    <li>linuxPackages_xanmod_rt.odp-dpdk</li>
    <li>linuxPackages_xanmod_rt.openrazer</li>
    <li>linuxPackages_xanmod_rt.pktgen</li>
    <li>linuxPackages_xanmod_rt.ply</li>
    <li>linuxPackages_xanmod_rt.r8168</li>
    <li>linuxPackages_xanmod_rt.rtl8188eus-aircrack</li>
    <li>linuxPackages_xanmod_rt.rtl8192eu</li>
    <li>linuxPackages_xanmod_rt.rtl8812au</li>
    <li>linuxPackages_xanmod_rt.rtl8814au</li>
    <li>linuxPackages_xanmod_rt.rtl8821au</li>
    <li>linuxPackages_xanmod_rt.rtl8821ce</li>
    <li>linuxPackages_xanmod_rt.rtl8821cu</li>
    <li>linuxPackages_xanmod_rt.rtl88x2bu</li>
    <li>linuxPackages_xanmod_rt.rtl88xxau-aircrack</li>
    <li>linuxPackages_xanmod_rt.rtw88 (linuxPackages_xanmod_rt.rtlwifi_new)</li>
    <li>linuxPackages_xanmod_rt.rtw89</li>
    <li>linuxPackages_xanmod_rt.stdenv</li>
    <li>linuxPackages_xanmod_rt.sysdig</li>
    <li>linuxPackages_xanmod_rt.system76</li>
    <li>linuxPackages_xanmod_rt.system76-acpi</li>
    <li>linuxPackages_xanmod_rt.system76-io</li>
    <li>linuxPackages_xanmod_rt.system76-power</li>
    <li>linuxPackages_xanmod_rt.systemtap</li>
    <li>linuxPackages_xanmod_rt.tmon</li>
    <li>linuxPackages_xanmod_rt.tp_smapi</li>
    <li>linuxPackages_xanmod_rt.turbostat</li>
    <li>linuxPackages_xanmod_rt.tuxedo-keyboard</li>
    <li>linuxPackages_xanmod_rt.usbip</li>
    <li>linuxPackages_xanmod_rt.v4l2loopback</li>
    <li>linuxPackages_xanmod_rt.v86d</li>
    <li>linuxPackages_xanmod_rt.veikk-linux-driver</li>
    <li>linuxPackages_xanmod_rt.vendor-reset</li>
    <li>linuxPackages_xanmod_rt.vhba</li>
    <li>linuxPackages_xanmod_rt.virtualbox</li>
    <li>linuxPackages_xanmod_rt.virtualboxGuestAdditions</li>
    <li>linuxPackages_xanmod_rt.x86_energy_perf_policy</li>
    <li>linuxPackages_xanmod_rt.xmm7360-pci</li>
    <li>linuxPackages_xanmod_rt.xpadneo</li>
    <li>linuxPackages_xanmod_rt.zenpower</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133511` at aa8f8c1d run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>47 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/9882ec828c8fd47ea4e27778610f9c8a">linuxPackages_xanmod_rt.acpi_call</a></li>
    <li><a href="https://gist.github.com/6544e472f4e884781cd32f31611c5ec9">linuxPackages_xanmod_rt.akvcam</a></li>
    <li><a href="https://gist.github.com/6ab7d6df30948d4093d8d9db90950232">linuxPackages_xanmod_rt.anbox</a></li>
    <li><a href="https://gist.github.com/c7be764550710024cd3a8aea23d2fa1d">linuxPackages_xanmod_rt.apfs</a></li>
    <li><a href="https://gist.github.com/27ea4411deefcaab6971b36891e0edb7">linuxPackages_xanmod_rt.batman_adv</a></li>
    <li><a href="https://gist.github.com/8aeec8a8ac98308947ed3250abe30372">linuxPackages_xanmod_rt.bcc</a></li>
    <li><a href="https://gist.github.com/fb157fa786c53b1962b3548ae7650054">linuxPackages_xanmod_rt.can-isotp</a></li>
    <li><a href="https://gist.github.com/90b963e4082ef199e0eae4b4a9a09ea7">linuxPackages_xanmod_rt.cryptodev</a></li>
    <li><a href="https://gist.github.com/ccf37bbf3326d2b2143f78701a63ba26">linuxPackages_xanmod_rt.ddcci-driver</a></li>
    <li><a href="https://gist.github.com/111de1dd38cca34edc86225579049793">linuxPackages_xanmod_rt.digimend</a></li>
    <li><a href="https://gist.github.com/ef99662a1cb349847884787df7290200">linuxPackages_xanmod_rt.ena</a></li>
    <li><a href="https://gist.github.com/3f0da208a5899719b7ecd00b23449c49">linuxPackages_xanmod_rt.evdi</a></li>
    <li><a href="https://gist.github.com/dc68f12e27daaf61efb561929074c344">linuxPackages_xanmod_rt.fwts-efi-runtime</a></li>
    <li><a href="https://gist.github.com/e60653d74362538e9309b1c46d15595d">linuxPackages_xanmod_rt.gcadapter-oc-kmod</a></li>
    <li><a href="https://gist.github.com/2a9961b571a715758c3351e3df295e06">linuxPackages_xanmod_rt.hid-nintendo</a></li>
    <li><a href="https://gist.github.com/3d3f93714edbd552fc6bce5d993063ba">linuxPackages_xanmod_rt.jool</a></li>
    <li><a href="https://gist.github.com/47cbe84a941ce24d2b656258d476bb88">linux_xanmod-rt</a> (linuxPackages_xanmod_rt.kernel)</li>
    <li><a href="https://gist.github.com/0aeac79117bcb36d60e82dcc9638693d">linuxPackages_xanmod_rt.lttng-modules</a></li>
    <li><a href="https://gist.github.com/8ec48a5b4a5db3fdd5b36d6907935b43">linuxPackages_xanmod_rt.mba6x_bl</a></li>
    <li><a href="https://gist.github.com/f79d6cc8ee3b7b706b0ef4cb415c6ef2">linuxPackages_xanmod_rt.mbp2018-bridge-drv</a></li>
    <li><a href="https://gist.github.com/1e9f992aeca9b08b20c6be64e8de5a59">linuxPackages_xanmod_rt.mwprocapture</a></li>
    <li><a href="https://gist.github.com/9fdca07edaf58432a15b8f4be483be9d">linuxPackages_xanmod_rt.mxu11x0</a></li>
    <li><a href="https://gist.github.com/5ffe1eacb766769d53055fcffb8b21f6">linuxPackages_xanmod_rt.netatop</a></li>
    <li><a href="https://gist.github.com/526e05511d47a978017e548bbbc1530f">linuxPackages_xanmod_rt.oci-seccomp-bpf-hook</a></li>
    <li><a href="https://gist.github.com/beb66aeda4e162c795466d14f6a435cf">linuxPackages_xanmod_rt.openafs</a> (linuxPackages_xanmod_rt.openafs_1_8)</li>
    <li><a href="https://gist.github.com/b9221b39a6c22e27abab1ad8c4e603bd">linuxPackages_xanmod_rt.openafs_1_9</a></li>
    <li><a href="https://gist.github.com/ecb8cbfec730f26354f49d254993bc6c">linuxPackages_xanmod_rt.openrazer</a></li>
    <li><a href="https://gist.github.com/5aacc812ca91900eae8843c2a1283210">linuxPackages_xanmod_rt.perf</a></li>
    <li><a href="https://gist.github.com/ad2821964bfa2a711e1612f028579c20">linuxPackages_xanmod_rt.r8125</a></li>
    <li><a href="https://gist.github.com/86c7e8c8af3b48a7f0d13a67bf8d7d34">linuxPackages_xanmod_rt.r8168</a></li>
    <li><a href="https://gist.github.com/72ef68cbd56ca32a107f486756a3a2b1">linuxPackages_xanmod_rt.rtl8192eu</a></li>
    <li><a href="https://gist.github.com/daaa8315eb36a601f170d026b37bb3f2">linuxPackages_xanmod_rt.rtl8723bs</a></li>
    <li><a href="https://gist.github.com/4fba0aff63580048d16b2a34cc03c281">linuxPackages_xanmod_rt.rtl8812au</a></li>
    <li><a href="https://gist.github.com/9d726a3b9998fa1c89e7939837106224">linuxPackages_xanmod_rt.rtl8821ce</a></li>
    <li><a href="https://gist.github.com/b55225b1ee91ba01bfc4a97864d21f73">linuxPackages_xanmod_rt.rtl8821cu</a></li>
    <li><a href="https://gist.github.com/2d80c34a9968ce772c51ce6cc7b95fcc">linuxPackages_xanmod_rt.rtl88x2bu</a></li>
    <li><a href="https://gist.github.com/c1487d301caa6882213e8b2cfd2ee9e4">linuxPackages_xanmod_rt.rtw88</a> (linuxPackages_xanmod_rt.rtlwifi_new)</li>
    <li><a href="https://gist.github.com/fc4428f6b86c4cafb50fbca979e1e771">linuxPackages_xanmod_rt.rtw89</a></li>
    <li><a href="https://gist.github.com/f5b3369d5ccbe59986eb745f6e67f3d9">linuxPackages_xanmod_rt.sch_cake</a></li>
    <li><a href="https://gist.github.com/0444499175b87f52bd9482339bb40cb3">linuxPackages_xanmod_rt.systemtap</a></li>
    <li><a href="https://gist.github.com/1b7834efb64124fc65f1e78d04f156cb">linuxPackages_xanmod_rt.tuxedo-keyboard</a></li>
    <li><a href="https://gist.github.com/48f79da81bcb9af7c2a83b34c705624d">linuxPackages_xanmod_rt.v4l2loopback</a></li>
    <li><a href="https://gist.github.com/65f236699939ec03bc95202373741067">linuxPackages_xanmod_rt.veikk-linux-driver</a></li>
    <li><a href="https://gist.github.com/867cbb229da6f575100e8017843b9f57">linuxPackages_xanmod_rt.vhba</a></li>
    <li><a href="https://gist.github.com/5fe4f80e64036a76aafb2a76c8331409">linuxPackages_xanmod_rt.xmm7360-pci</a></li>
    <li><a href="https://gist.github.com/89541d64a8afbc07e5662e5aaf75e62f">linuxPackages_xanmod_rt.xpadneo</a></li>
    <li><a href="https://gist.github.com/61119cfdd98a817e8eb037c62b917457">linuxPackages_xanmod_rt.zfs</a> (linuxPackages_xanmod_rt.zfsStable ,linuxPackages_xanmod_rt.zfsUnstable)</li>
  </ul>
</details>
<details>
  <summary>1 package skipped due to time constraints:</summary>
  <ul>
    <li>linuxPackages_xanmod_rt.dpdk-kmods</li>
  </ul>
</details>
<details>
  <summary>6 packages built successfully:</summary>
  <ul>
    <li>linuxPackages_xanmod_rt.cpupower</li>
    <li>linuxPackages_xanmod_rt.hyperv-daemons</li>
    <li>linuxPackages_xanmod_rt.odp-dpdk</li>
    <li>linuxPackages_xanmod_rt.stdenv</li>
    <li>linuxPackages_xanmod_rt.tmon</li>
    <li>linuxPackages_xanmod_rt.usbip</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Review by SuperSandro2000 — COMMENTED** [review]
Isn't this an almost exact copy of xanmod? Can't we reduce the package file from there?

**Comment by fortuneteller2k** [comment]
@ofborg eval linux_xanmod

**Comment by fortuneteller2k** [comment]
@ofborg eval linux_xanmod

File: pkgs/os-specific/linux/kernel/linux-xanmod.nix [hunk 1]
```diff
@@ -1,9 +1,9 @@
-{ lib, stdenv, buildLinux, fetchFromGitHub, ... } @ args:
+{ lib, stdenv, buildLinux, fetchFromGitHub, realtime ? false, ... } @ args:
 
 let
-  version = "5.13.12";
+  version = if !realtime then "5.13.12" else "5.13.1";
   release = "1";
-  suffix = "xanmod${release}-cacule";
+  suffix = if !realtime then "xanmod${release}-cacule" else "rt${release}-xanmod${release}";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/kernel/linux-xanmod.nix:6]: That is not exactly what I had in mind. I think boost and php are good examples for packages which share common files. By using inputs for different things and supplying them at a common place we can avoid the if else around the file and still re-use common parts.

**Pull request closed without merging**